src/common/types.cpp

```diff
@@ -30,12 +30,42 @@
 #include <sstream>
 #include <sys/socket.h>
 
+#include <openthread/dataset.h>
+
+#include "common/byteswap.hpp"
 #include "common/code_utils.hpp"
 #include "common/logging.hpp"
 #include "common/types.hpp"
 
 namespace otbr {
 
+std::string BytesToHexString(const uint8_t *aHex, uint16_t aLength)
+{
+    std::string str;
+    char        bytestr[3];
+
+    for (uint16_t i = 0; i < aLength; i++)
```

Comment:
```suggestion
    str.reserve(aLength * 2);

    for (uint16_t i = 0; i < aLength; i++)
```
use [string::reserve](https://www.cplusplus.com/reference/string/string/reserve/) for efficiency